pallets/teerex/src/tests/test_cases.rs

```diff
@@ -1026,3 +1026,23 @@ fn publish_hash_with_too_much_data_fails() {
 		);
 	})
 }
+
+#[test]
+fn add_enclave_works_exceeds_number() {
+	new_test_ext().execute_with(|| {
+		crate::EnclaveCountMax::<Test>::put(0u64);
+		assert_eq!(Teerex::enclave_count_max(), 0);
+		Timestamp::set_timestamp(TEST5_TIMESTAMP);
+		let signer5 = get_signer(TEST5_SIGNER_PUB);
+		assert_err!(
+			Teerex::register_enclave(
+				RuntimeOrigin::signed(signer5),
+				TEST5_CERT.to_vec(),
+				URL.to_vec(),
+				None,
+				None,
+			),
+			Error::<Test>::ExceedEnclaveNumber
+		);
+	})
```

Comment:
I think we also have to make sure the enclave registration succeeds if we set the `MaxEnclaveCount` to 1 or higher